kafka-rest/src/test/java/io/confluent/kafkarest/controllers/ClusterManagerImplTest.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.kafkarest.controllers;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static org.easymock.EasyMock.anyObject;
+import static org.easymock.EasyMock.expect;
+import static org.easymock.EasyMock.replay;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.fail;
+
+import io.confluent.kafkarest.entities.Broker;
+import io.confluent.kafkarest.entities.Cluster;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.DescribeClusterResult;
+import org.apache.kafka.common.KafkaFuture;
+import org.apache.kafka.common.Node;
+import org.apache.kafka.common.errors.TimeoutException;
+import org.easymock.EasyMockRule;
+import org.easymock.Mock;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class ClusterManagerImplTest {
+
+  private static final String CLUSTER_ID = "cluster-1";
+  private static final Node NODE_1 = new Node(1, "broker-1", 9091);
+  private static final Node NODE_2 = new Node(2, "broker-2", 9092);
+  private static final Node NODE_3 = new Node(3, "broker-3", 9093);
+  private static final List<Node> NODES = Arrays.asList(NODE_1, NODE_2, NODE_3);
+
+  @Rule
+  public final EasyMockRule mocks = new EasyMockRule(this);
+
+  @Mock
+  private Admin adminClient;
+
+  @Mock
+  private DescribeClusterResult describeClusterResult;
+
+  private ClusterManagerImpl clusterManager;
+
+  @Before
+  public void setUp() {
+    clusterManager = new ClusterManagerImpl(adminClient);
+  }
+
+  @Test
+  public void listClusters_returnListWithOwnCluster() throws Exception {
+    expect(adminClient.describeCluster(anyObject())).andReturn(describeClusterResult);
```

Comment:
Why do we specify in the test name - `returnListWithOwnCluster` ? Don't we simply list all the clusters ?

Reply:
So, there's catch currently which is that Kafka (and the REST Proxy) does not know about other clusters. It only knows about the Cluster pointed by the `bootstrap.servers` property. I created the API this way so that in the future, when/if we do have information about other clusters, we can return them in the list as well. But right now, listing clusters always returns only one cluster.

Reply:
Okay, thank you.